File: resources/busca_binaria/busca_binaria.c
int binarySearch(int arr[], int l, int r, int x)
{
    while (l <= r) {
        int m = (r + l) / 2;
        if (arr[m] >= x) {
            if (x >= arr[m]) {
                return m;
            } else {
                r = m - 1;
            }
        } else {
            l = m + 1;
        }

        // if (arr[m] == x) return m;
        // if (arr[m] > x) r = m - 1;
        // if (arr[m] < x) l = m + 1;
    }
    return -1;
}

File: resources/busca_binaria/busca_binaria.py
def busca_binaria(arr, l, r, x):
    ans = -1

    if l > r:
        ans = -1
        return ans

    meio = (r + l) / 2;

    if arr[meio] == x:
        ans = l
        return ans

    if arr[meio] == x:
        ans = r
        return ans

    busca_binaria(arr, meio + 1, r, x)


if __name__ == "__main__":
    print(busca_binaria(10000))
